src/Util.vala

```diff
@@ -107,26 +107,39 @@ namespace Ilia {
         // Normalize strings for comparison
         string s_lower = source.down();
         string p_lower = pattern.down();
+    
+        // Perfect exact match
+        if (s_lower == p_lower) {
+            return 100;
+        }
         
         // Check for exact substring match first (highest priority)
         if (s_lower.contains(p_lower)) {
-            // Calculate score based on match position and completeness
-            // Give higher scores to matches at the beginning
             int pos = s_lower.index_of(p_lower);
             if (pos == 0) {
-                // Starting match is best
+                // Starting match is best - high score
                 return 95 + int.min(5, 5 * pattern.length / source.length);
             } else {
-                // Penalize matches later in the string
-                return 80 + int.min(15, 20 * pattern.length / source.length) - int.min(15, pos / 2);
+                if (pos > 0 && (s_lower[pos - 1] == ' ' || s_lower[pos - 1] == '-' || s_lower[pos - 1] == '_')) {
+                    return 85 + int.min(10, 10 * pattern.length / source.length);
+                }
+                // Regular substring match - penalize based on position
+                return 75 + int.min(10, 15 * pattern.length / source.length) - int.min(10, pos / 3);
```

Comment:
Refer to my comment on the old commit. See if you can generalize these cases and make the formula more readable. Also is there a specific reason for prioritizing cases with _,- or space?